File: beacon_monitoring/android/src/main/kotlin/uk/co/objectivity/beacon_monitoring/plugin/model/RangingResult.kt
// Copyright (c) 2020 Objectivity. All rights reserved.
// Use of this source code is governed by The MIT License (MIT) that can be
// found in the LICENSE file.

package uk.co.objectivity.beacon_monitoring.plugin.model

import com.squareup.moshi.JsonClass
import com.squareup.moshi.Moshi

fun RangingResult.toJson(): String = Moshi.Builder().build().adapter(RangingResult::class.java).toJson(this)

@JsonClass(generateAdapter = true)
data class RangingResult(val region: RegionModel, val beacons: List<BeaconModel>)

File: beacon_monitoring_platform_interface/lib/src/model/ranging_result.dart
// Copyright (c) 2020 Objectivity. All rights reserved.
// Use of this source code is governed by The MIT License (MIT) that can be
// found in the LICENSE file.

import 'dart:convert';

import 'package:flutter/foundation.dart';

import 'region.dart';
import 'beacon.dart';

class RangingResult {
  static const _region = 'region';
  static const _beacons = 'beacons';

  final Region region;
  final List<Beacon> beacons;

  RangingResult({
    @required this.region,
    @required this.beacons,
  });

  RangingResult.fromJson(Map<String, dynamic> json)
      : region = Region.fromJson(json[_region]),
        beacons = json[_beacons] != null
            ? (json[_beacons] as List<dynamic>)
                .map((it) => Beacon.fromJson(it as Map<String, dynamic>))
                .toList()
            : null;

  Map<String, dynamic> toJson() => <String, dynamic>{
        _region: region.toJson(),
        _beacons: beacons.map((e) => e.toJson()).toList(),
      };

  @override
  String toString() => jsonEncode(toJson());
}
